omc3/check_corrections.py

```diff
@@ -0,0 +1,612 @@
+    """ Sort the given correction files:
+    If given by individual files, they all go into one bucket,
+    if given by folders (i.e. scenarios) they are sorted by its name.
+    It is also checked, that they are valid!"""
+    # create correction mapping
+    if corrections[0].is_file():  # checked above, that all are files or all are dirs
+        corr_dict = {"": corrections}
+    else:
+        corr_dict = {c.name:  _glob_regex_paths(c, file_pattern) for c in corrections}
+
+    # check correction files
+    for name, corr_files in corr_dict.items():
+        if not len(corr_files):
+            raise IOError(f"No corrections found for scenario {name}.")
+
+        do_not_exist = [f for f in corr_files if not f.exists()]
+        if len(do_not_exist):
+            raise IOError(f"Some correction files do not exist for scenario {name}:"
+                          f" {str(do_not_exist)}")
+
+    return corr_dict
+
+
+def _glob_regex_paths(path: Path, pattern: str) -> List[Path]:
+    """ Filter the files in path by pattern and return a list of paths. """
+    return [path / f for f in glob_regex(path, pattern)]
+
+
+# Main and Output --------------------------------------------------------------
+
+def _get_measurement_filter(nominal_model: TfsDataFrame, opt: DotDict) -> Dict[str, pd.Index]:
+    """ Writes the filtered original data out, so that it can be easily plotted against
+    the corrections later. We now have the same BPMs and Column-Names. """
+    if not opt.optics_params:
+        LOG.debug("No filters selected, returning empty dict.")
+        return {}
+
+    if TUNE in opt.optics_params:
+        LOG.warning("Filtering RMS on tune does not make sense. Ignoring.")
+        opt.optics_params.remove(TUNE)
+
+    optics_params, meas_dict = global_correction.get_measurement_data(
+        opt.optics_params,
+        opt.meas_dir,
+        opt.beta_filename,
+    )
+
+    # add weights and use_errorbars as these are required in filters
+    opt.use_errorbars = False
+    opt.weights = {param: 1.0 for param in opt.optics_params}
+    meas_dict = filters.filter_measurement(optics_params, meas_dict, nominal_model, opt)
+
+    # use the indices as filters
+    filter_dict = {
+        global_correction.get_filename_from_parameter(k, opt.beta_filename): v.index for k, v in meas_dict.items()
+    }
+
+    LOG.debug(f"Filters selected for keys: {list(filter_dict.keys())}")
+    return filter_dict
+
+
+def _create_model_and_write_diff_to_measurements(
+        output_dir: Path, measurement: OpticsMeasurement, correction_name: str, correction_files: Sequence[Path],
+        accel_inst: Accelerator, rms_masks: Dict) -> OpticsMeasurement:
+    """ Create a new model with the corrections (well, the "matchings") inserted and calculate
+    the difference to the measurements. This will be written out then into individual tfs-files in the output
```

Comment:
I would also specify that we calculate the difference to the nominal model and then apply this difference on the measurement to "predict" the effect of the correction on the measured machine (if I am correct about that), not that we calculate a difference from matched model to measurements as the sentence implies.